src/swarmus_ros_simulation/launch/reusable/simulated_agent.launch

```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" ?>
+<!-- This launch files spawns the required nodes for a robot to be operational as a swarm agent. These include:
+    - The HiveMind connector node
+    - The HiveMind node itself
+    - The HiveConnect node
+    - The navigation stack mode
+-->
+<launch>
+    <!-- The following arguments are used to the different parameters of each nodes. -->
+
+    <!-- Generic parameters -->
+    <arg name="agent_id"   default="1"/>
+
+    <!-- Navigation stack parameters -->
+    <arg name="cmd_vel_topic" default="cmd_vel"/>
+    <arg name="odom_topic"    default="odom"/>
+
+    <!-- HiveMind Connector Node -->
+    <arg name="connector_server_port" default="$(eval 7000+arg('agent_id'))"/>
+    <arg name="connector_robot_name_prefix" default="pioneer_$(arg agent_id)"/>
+
+    <!-- HiveMind Node -->
+    <arg name="hivemind_board_uuid" default="$(arg agent_id)"/>
+    <arg name="hivemind_host_port" default="$(eval 7000 + arg('agent_id'))"/>
+    <arg name="hivemind_remote_port_mock" default="$(eval 8000 + arg('agent_id'))" />
+
+    <!-- HiveConnect Node -->
+    <arg name="hive_connect_spi_mock_port" default="$(eval 8000 + arg('agent_id'))"/>
+    <arg name="hive_connect_listen_port" default="$(eval 9000 + arg('agent_id'))"/>
+
+    <!-- Starting the nodes after arguments definition -->
+    <!-- Bridge -->
+    <node pkg="hive_mind_bridge" type="hive_mind_bridge" name="hive_mind_bridge_$(arg agent_id)" output="screen">
```

Comment:
Instead of adding an argument in the node's name directly to give a different name for each agents, you can used namespaces in launchfile. 
```
<group ns="$(arg agent)">
     <node pkg="hive_mind_bridge" type="hive_mind_bridge" name="hive_mind_bridge" output="screen">
     ...
</group>
```
This will add a prefix "/agent_id" to the node's name (/agent_id/hive_mind_bridge) and all topics published by the node will have this prefix to.

I'vs just realised that on line 77 to 79, the robot_name is only used to add that kind of prefix:
```
    std::string robotName = ros::param::param("~ROBOT_NAME", std::string("pioneer_0"));
    ros::Publisher moveByPublisher = nodeHandle.advertise<swarmus_ros_navigation::MoveByMessage>(
        robotName + "/navigation/moveBy", 1000);
```
So you could actually remove the robot_name param from this node and only use the namespace feature of launchfiles.